packages/runtime/src/reflection-contains-message-type.ts

```diff
@@ -0,0 +1,22 @@
+import {IMessageType, MESSAGE_TYPE} from './message-type-contract';
+
+/**
+ * The interface that models storing type in a symbol property.
+ */
+export interface MessageTypeContainer<T extends object> {
+    [MESSAGE_TYPE]: IMessageType<T>;
+}
+
+/**
+ * Check if the provided object is a proto message.
+ */
+export function containsMessageType<T extends object>(msg: object): msg is MessageTypeContainer<T> {
+    return (msg as MessageTypeContainer<T>)[MESSAGE_TYPE] != null;
+}
+
```

Comment:
The typing I suggested originally was incomplete. This will work:

```suggestion
export function containsMessageType<T extends object>(msg: T): msg is (T & MessageTypeContainer<T>) {
    return (msg as MessageTypeContainer<T>)[MESSAGE_TYPE] != null;
}

```

It let's the compiler know that `msg` has the `MESSAGE_TYPE` property:

```ts
if (containsMessageType(msg)) {
     msg[MESSAGE_TYPE].toJson(msg)
}
```

It will still let you access regular properties of `msg` and also works on `UnknownMessage`.

Reply:
👍